### PR TITLE
feat(rule): Create NeglectedAlert table

### LINKED ISSUE #56190: [BE] Write a migration to create the new `NeglectedAlert` table
- Make a new table `NeglectedAlert`
    - `rule_id` nullable False
    - `org_id`
    - `disable_date` will be “now” (when the script is run) + 14 days nullable False
    - `opted_out` boolean so when we re-run we don’t grab the same alert default False
    - `has_sent_initial_email` boolean default False
    - `has_sent_final_email` boolean default False

### PR DESCRIPTION
**Pull request by ceorourke**
Create a new `NeglectedAlert` table to store data about rules that haven't been clicked on in X period of time to schedule them for a migration that will disable the rule. 

Spec [here](https://www.notion.so/sentry/Disable-Neglected-Alerts-500e904502944d8e963fcd8a693e38bd) for more context.

Closes #56190

### CODE REVIEW

**Comment by ceorourke** [comment]
Tagging @getsentry/issues for visibility - will not merge until I get approval from @getsentry/owners-migrations

File: src/sentry/models/rule.py
Comment (scefali) [inline src/sentry/models/rule.py:148]: @ceorourke we likely want indexes on some of these columns but we can wait until you figure out what the query looks like to add them

File: src/sentry/models/rule.py
Comment (scefali) [inline src/sentry/models/rule.py:144]: @ceorourke I think we should add the `org_id` as well to make it easier to query the organization

File: src/sentry/models/rule.py
Comment (scefali) [inline src/sentry/models/rule.py:140]: @ceorourke i think having this in the region silo makes this more difficult. You have to run the migration per-region this way. I think what we need is to run it in the control silo.

File: src/sentry/migrations/0552_create_neglectedalert_table.py
Comment (markstory) [inline src/sentry/migrations/0552_create_neglectedalert_table.py:39]: I would be careful with this. You can never move or rename that function as long as this migration lives.

File: src/sentry/models/rule.py
Comment (markstory) [inline src/sentry/models/rule.py:136]: Maybe leave a comment that this is used in migrations and can't be removed/moved.

**Review by markstory — APPROVED** [review]
Schema looks good to me. If you're going to query on the has_ fields and this table gets big, you'll want an index on those fields.

File: src/sentry/models/rule.py
Comment (ceorourke) [reply src/sentry/models/rule.py:136]: @markstory if for some reason we need to change it, we edit the function itself right? It'd just be weird to have `in_fourteen_days` set 30 days (or some other number)?

Comment (ceorourke) [reply src/sentry/models/rule.py:136]: Thought about it more and I'm just gonna remove it - I was doing it to make it a teeny bit easier when populating to not have to find 14 days in the future but it's not a big deal.

File: src/sentry/models/rule.py
Comment (scefali) [inline src/sentry/models/rule.py:143]: @ceorourke Nit: could we put `date` somewhere in these column names?

**Review by scefali — APPROVED** [review]

**Pull request merged**
